jobs/mysql_restore_job.go

```diff
@@ -167,30 +167,29 @@ func restoreMySQLFromS3(
 		return errors.Wrapf(err, "stderr: %s\n stdout: %s\n", stderr.String(), stdout.String())
 	}
 
+	xbcloudErr := make(chan error)
 	if err := xbcloudCmd.Start(); err != nil {
-		return errors.Wrap(wrapError(err), "xbcloud start failed")
+		return wrapError(err)
 	}
-	defer func() {
-		err := xbcloudCmd.Wait()
-		if err == nil {
-			return
-		}
 
-		if rerr != nil {
-			rerr = errors.Wrapf(rerr, "xbcloud wait error: %s", err)
-		} else {
-			rerr = errors.Wrap(wrapError(err), "xbcloud wait failed")
-		}
+	go func() {
+		xbcloudErr <- xbcloudCmd.Wait()
```

Comment:
the documentation says it is incorrect to call Wait before all reads from the pipe have been completed.
https://pkg.go.dev/os/exec#Cmd.StdoutPipe

Probably a better option is to use io.Pipe